src/common/Thread.cc

```diff
@@ -31,6 +33,8 @@
 #include <sched.h>
 #endif
 
+// Thread Local Storage
+thread_local unsigned int global_thread_id = 0;
```

Comment:
C++ allows for `thread_local` static class member while the ID (or cached thread name) conceptually belongs to `Thread`.

The _global_ in the name is bit misleading IMHO.

Reply:
hmm